### PR TITLE
move vmware extension here

### PR DESCRIPTION
**Pull request by ctaggart**
This is for moving az-vmware-cli to this repo. I just released https://github.com/Azure/az-vmware-cli/releases/tag/0.7.0. This is a copy of that code. Once it is imported, reviewed, and tested a bit more, we'd like a 1.0.0 published and put in the index.json.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
vmware

**Comment by yungezz** [comment]
HI @fengzhou-msft could  you pls help to review? This is the PR to move VMware extension from their own repo to CLI extension repo.

**Comment by ctaggart** [comment]
Looks like the static analysis failure was just formatting. I'll have the fixed shortly.
```
/home/vsts/work/1/s/src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py:113:1: W391 blank line at end of file
2     E128 continuation line under-indented for visual indent
1     E225 missing whitespace around operator
1     E261 at least two spaces before inline comment
24    E302 expected 2 blank lines, found 1
1     E305 expected 2 blank lines after class or function definition, found 1
16    W291 trailing whitespace
4     W292 no newline at end of file
2     W293 blank line contains whitespace
6     W391 blank line at end of file
```

**Comment by ctaggart** [comment]
After getting azdev setup, I was able to reproduce the formatting errors with:
```
azdev extension build vmware
azdev style vmware
```

Is there a way to have the code formatted automatically? I tried installing and running [black](https://pypi.org/project/black/), but the formatting does not match what `azdev style` is checking.

**Comment by fengzhou-msft** [comment]
> After getting [azdev setup](https://github.com/Azure/azure-cli-dev-tools), I was able to reproduce the formatting errors with:
> 
> ```
> azdev extension build vmware
> azdev style vmware
> ```
> 
> Is there a way to have the code formatted automatically? I tried installing and running [black](https://pypi.org/project/black/), but the formatting does not match what `azdev style` is checking.

`azdev style` uses `pylint` and `flake8`. It may have some customized rules. @haroldrandom any suggestions?

**Comment by ctaggart** [comment]
I am headed out on leave for 12 weeks now, which is why we want this extension moved in here. Hopefully what I've provided is enough.

**Comment by haroldrandom** [comment]
@ctaggart there is some gaps betweeen pylint's rules/flake8's rules and black's rules, can't just black the python file.

@haroldrandom the customized rules are palced in pylintrc and .flake8.

**Comment by qwordy** [comment]
You have some style errors in code according the checks.

**Comment by ctaggart** [comment]
The metadata changes need to be brought in here from https://github.com/Azure/az-vmware-cli/pull/43

**Comment by ctaggart** [comment]
I was able to reproduce the flake8 errors with `flake8.exe --config=.flake8 .\src\vmware\azext_vmware\`. They appear to be fixed now. Any idea what I need to run locally in order to reproduce these pylint errors:
```
Run custom pylint rules.
Running pylint on extensions...
************* Module azext_vmware.commands
src/vmware/azext_vmware/commands.py:18:8: E5001: Show command must use show_command or custom_show_command. (show-command)
src/vmware/azext_vmware/commands.py:32:8: E5001: Show command must use show_command or custom_show_command. (show-command)
src/vmware/azext_vmware/commands.py:38:8: E5001: Show command must use show_command or custom_show_command. (show-command)
src/vmware/azext_vmware/commands.py:44:8: E5001: Show command must use show_command or custom_show_command. (show-command)
```

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -154,3 +154,4 @@
 
 /src/stack-hci/ @zhoxing-ms
 
+/src/vmware/ @fengzhou-msft @qwordy
```
Comment (ctaggart) [inline .github/CODEOWNERS:157]: cc @fengzhou-msft @qwordy

**Comment by ctaggart** [comment]
@fengzhou-msft & @qwordy, I fixed the lint errors. Can you guys take it from here? Since AVS is GA, we really want to get this handed off, make this a 1.0.0, and get it published.

**Comment by qwordy** [comment]
Some checks were not successful
```
ERROR: test_vmware (azext_vmware.tests.latest.test_vmware_scenario.VmwareScenarioTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/azure/cli/core/commands/arm.py", line 754, in handler
    return getter(**args)
  File "/home/vsts/work/1/s/src/vmware/azext_vmware/custom.py", line 18, in privatecloud_show
    return client.private_clouds.get(resource_group_name, name)
  File "/home/vsts/work/1/s/src/vmware/azext_vmware/vendored_sdks/operations/private_clouds_operations.py", line 219, in get
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: ResourceNotFound
Message: The Resource 'Microsoft.AVS/privateClouds/cloud1' under resource group 'cli_test_vmware000001' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
```

**Comment by ctaggart** [comment]
@qwordy, that is an integration test and you need access to AVS or an AVS test environment to run. Can we disable it somehow here, since it isn't a unit test?

**Comment by qwordy** [comment]
> 
> 
> @qwordy, that is an integration test and you need access to AVS or an AVS test environment to run. Can we disable it somehow here, since it isn't a unit test?

Yes, you can. But make sure the code can run.

**Comment by ctaggart** [comment]
I figured out what was wrong with the tests. I ran it live and made a new test recording. Switching to `.show_command` throws a different type of error. The only thing failing is "Verify Ref Docs". The error I see is:
```
AssertionError: Request to https://azurecliprod.blob.core.windows.net/cli-extensions/guestconfig-0.1.0-py3-none-any.whl failed with 500
```
I'm not sure what that means. Looks like a CI error.

**Comment by qwordy** [comment]
I can open the link. I think it is a temporary network connection issue. Let's rerun it.

**Comment by ctaggart** [comment]
> I can open the link. I think it is a temporary network connection issue. Let's rerun it.

@qwordy, great, it worked this time! If this is going to take a few days to get released, can you please merge #2546 today, while this gets worked on? We have customers doing a `az extension add --name vmware` and getting the `0.6.0` build which uses the preview API, which has known bugs.

**Comment by qwordy** [comment]
> 
> 
> > I can open the link. I think it is a temporary network connection issue. Let's rerun it.
> 
> @qwordy, great, it worked this time! If this is going to take a few days to get released, can you please merge #2546 today, while this gets worked on? We have customers doing a `az extension add --name vmware` and getting the `0.6.0` build which uses the preview API, which has known bugs.

It is merged now.

File: src/vmware/setup.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from codecs import open
+from setuptools import setup, find_packages
+
+VERSION = "1.0.0"
```
Comment (qwordy) [inline src/vmware/setup.py:11]: Is the version correct?

**Review by qwordy — APPROVED** [review]

**Comment by qwordy** [comment]
@fengzhou-msft

**Comment by ctaggart** [comment]
@fengzhou-msft & @qwordy, can this get merged soon?

**Pull request merged**